controllers/grid/issues/form/IssueForm.php

```diff
@@ -241,7 +241,7 @@ public function execute(...$functionArgs)
         $issue->setVolume(empty($volume) ? null : $volume);
         $issue->setNumber(empty($number) ? null : $number);
         $issue->setYear(empty($year) ? null : $year);
-        if (!$isNewIssue) {
+        if ($this->getData('datePublished')) {
```

Comment:
Does this prevent the date published from being removed once it's been added through the form?

Reply:
You are thinking of a scenario where a new issue is added and the date is set, but the issue not published and the user wants to use remove it altogether?

Reply:
Testing steps:

1. Open the form and put `2022-02-03` into the date published field.
2. Click save.
3. Close form and re-open form.
4. Delete `2022-02-03` from the date published field.
5. Click save.
6. Close form and re-open form.

The date published field should be empty.

Reply:
So this is what I am thinking of adding

 ```
       // If issue is not published, allow empty datePublished
        if (!$this->getData('datePublished') && !$issue->getPublished()){
            $issue->setDatePublished(NULL);
        }

        // If datePublished is set
        if ($this->getData('datePublished')) {
            $issue->setDatePublished($this->getData('datePublished'));
        }
```

Reply:
If it works with the steps above, ok by me. What happens when an issue is published, though? Is there any error message when they try to edit the date published? Or does it just not change silently?

Reply:
Note that this does not allow you to give an empty value for an issue that is published.